pasture-algorithms/src/convexhull.rs

```diff
@@ -0,0 +1,596 @@
+use std::hash::{Hash, Hasher};
+use std::collections::HashSet;
+use pasture_core::{
+    containers::{PointBuffer, PointBufferExt},
+    layout::attributes::{POSITION_3D},
+    nalgebra::{Vector3},
+};
+
+#[derive(Clone, Copy)]
+struct Triangle {
+    a: usize,
+    b: usize,
+    c: usize,
+    normal: Vector3<f64>,
+}
+
+#[derive(Eq, Clone, Copy)]
+struct Edge {
+    a: usize,
+    b: usize,
+}
+
+impl PartialEq for Edge {
+    fn eq(&self, other: &Self) -> bool {
+        self.a == other.a && self.b == other.b ||
+        self.a == other.b && self.b == other.a
+    }
+}
+
+impl Hash for Edge {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        (self.a * self.b).hash(state);
+    }
+}
+
+/// Convex Hull generation as triangle mesh
+/// Returns the convex hull as a vector of tuples of size 3 that contains the indices of the triangle vertices within the input buffer that form a convex hull around all input points
+/// or an error if less than 3 linearily independent points were given in the input buffer.
+pub fn convex_hull_as_triangle_mesh<T: PointBuffer>(buffer: &T) -> Result<Vec<(usize, usize, usize)>, &'static str> {
```

Comment:
In 2c5d8e1 i changed this to use return a `Vector3<usize>` as you suggested. I decided against introducing a new struct since it would serve no functional purpose in this case.